File: app/src/main/java/org/macho/beforeandafter/record/ImageCache.java
package org.macho.beforeandafter.record;

import android.graphics.Bitmap;
import android.util.LruCache;

/**
 * Created by yuukimatsushima on 2017/10/21.
 */

public class ImageCache {
    private LruCache<String, Bitmap> lruCache;

    public ImageCache() {
        long maxMemory = Runtime.getRuntime().maxMemory() / 1024;
        int cacheSize = (int) (maxMemory / 8);

        lruCache = new LruCache<String, Bitmap>(cacheSize) {
            protected int sizeOf(String key, Bitmap value) {
                return value.getAllocationByteCount() / 1024;
            }
        };
    }

    public void put(String key, Bitmap image) {
        lruCache.put(key, image);
    }

    public Bitmap get(String key) {
        return lruCache.get(key);
    }

    public void clear() {
        lruCache.evictAll();
    }
}

File: app/src/main/java/org/macho/beforeandafter/record/ImageCache.kt
package org.macho.beforeandafter.record

import android.graphics.Bitmap
import android.util.LruCache

class ImageCache {
    private var lruCache: LruCache<String, Bitmap>

    init {
        val maxMemory = Runtime.getRuntime().maxMemory() / 1024
        val cacheSize = (maxMemory / 8).toInt()

        lruCache = object: LruCache<String, Bitmap>(cacheSize) {
            override fun sizeOf(key: String?, value: Bitmap?): Int {
                if (value == null) {
                    return 0
                }
                return value.allocationByteCount / 1024
            }
        }
    }


    fun put(key: String, image: Bitmap) {
        lruCache.put(key, image)
    }

    fun get(key: String): Bitmap? {
        return lruCache.get(key)
    }

    fun clear() {
        lruCache.evictAll()
    }
}